src/app/wallets/update-pending-invoices.ts

```diff
@@ -239,65 +255,79 @@ export const declineHeldInvoice = async ({
   pubkey: Pubkey
   paymentHash: PaymentHash
   logger: Logger
-}): Promise<boolean | ApplicationError> => {
-  const lndService = LndService()
-  if (lndService instanceof Error) return lndService
-
-  const walletInvoicesRepo = WalletInvoicesRepository()
-
-  const lnInvoiceLookup = await lndService.lookupInvoice({ pubkey, paymentHash })
-
-  const pendingInvoiceLogger = logger.child({
-    hash: paymentHash,
-    pubkey,
-    lnInvoiceLookup,
-    topic: "payment",
-    protocol: "lightning",
-    transactionType: "receipt",
-    onUs: false,
-  })
-
-  if (lnInvoiceLookup instanceof InvoiceNotFoundError) {
-    const isDeleted = await walletInvoicesRepo.deleteByPaymentHash(paymentHash)
-    if (isDeleted instanceof Error) {
-      pendingInvoiceLogger.error("impossible to delete WalletInvoice entry")
-      return isDeleted
-    }
-    return false
-  }
-  if (lnInvoiceLookup instanceof Error) return lnInvoiceLookup
-
-  // FIXME: This is just to support transition to hodl invoices
-  // TODO: REMOVE THIS after hodl invoices has been deployed for 24 hours.
-  if (lnInvoiceLookup.isSettled) {
-    addAttributesToCurrentSpan({ ["isHodlInvoice"]: false })
-    const walletInvoice = await WalletInvoicesRepository().findByPaymentHash(paymentHash)
-    if (walletInvoice instanceof Error) return walletInvoice
-    return updatePendingInvoice({ walletInvoice, logger })
-  }
-  addAttributesToCurrentSpan({ ["isHodlInvoice"]: true })
+}): Promise<boolean | ApplicationError> =>
+  asyncRunInSpan(
```

Comment:
`declineHeldInvoice` is already wrapped please only add missing attributes

Reply:
Are you sure? The prompt for this PR was the `isHodlInvoice` attribute being assigned to the parent `handleHeldInvoices` span resulting in us losing the granularity from individual runs of this function -> https://ui.honeycomb.io/galoy/datasets/galoy-bbw/result/eaYcthFH7Uj

Reply:
Made some selective-wrapping changes in trigger from here: https://github.com/GaloyMoney/galoy/pull/1526/commits/c215222764e2f1b83f309853366bbebb89021e02